packages/global-styles/src/components/global-styles-variations/index.tsx

```diff
@@ -0,0 +1,135 @@
+	premiumBadge?: React.ReactNode;
+	showOnlyHoverView?: boolean;
+	onSelect: () => void;
+}
+
+interface GlobalStylesVariationsProps {
+	globalStylesVariations: GlobalStylesObject[];
+	selectedGlobalStylesVariation: GlobalStylesObject | null;
+	premiumBadge?: React.ReactNode;
+	showOnlyHoverViewDefaultVariation?: boolean;
+	onSelect: ( globalStylesVariation: GlobalStylesObject ) => void;
+}
+
+const isDefaultGlobalStyleVariationSlug = ( globalStylesVariation: GlobalStylesObject ) =>
+	globalStylesVariation.slug === DEFAULT_GLOBAL_STYLES_VARIATION_SLUG;
+
+const GlobalStylesVariation = ( {
+	globalStylesVariation,
+	isActive,
+	premiumBadge,
+	showOnlyHoverView,
+	onSelect,
+}: GlobalStylesVariationProps ) => {
+	const [ isFocused, setIsFocused ] = useState( false );
+	const { base } = useContext( GlobalStylesContext );
+	const context = useMemo( () => {
+		return {
+			user: globalStylesVariation,
+			base,
+			merged: mergeBaseAndUserConfigs( base, globalStylesVariation ),
+		};
+	}, [ globalStylesVariation, base ] );
+
+	const selectOnEnter = ( event: React.KeyboardEvent ) => {
+		if ( event.keyCode === ENTER ) {
+			event.preventDefault();
+			onSelect();
+		}
+	};
+
+	return (
+		<div
+			className={ classnames( 'global-styles-variations__item', {
+				'is-active': isActive,
+			} ) }
+			role="button"
+			onBlur={ () => setIsFocused( false ) }
+			onFocus={ () => setIsFocused( true ) }
+			onClick={ onSelect }
+			onKeyDown={ selectOnEnter }
+			tabIndex={ 0 }
+			aria-current={ isActive }
+			aria-label={
+				translate( 'Style: %s', {
+					comment: 'Aria label for style preview buttons',
+					args: globalStylesVariation.title,
+				} ) as string
+			}
+		>
+			{ premiumBadge }
+			<div className="global-styles-variation__item-preview">
+				<GlobalStylesContext.Provider value={ context }>
+					<GlobalStylesVariationPreview
+						title={ globalStylesVariation.title }
+						isFocused={ isFocused || showOnlyHoverView }
+					/>
+				</GlobalStylesContext.Provider>
+			</div>
+		</div>
+	);
+};
+
+const GlobalStylesVariations = ( {
+	globalStylesVariations,
+	selectedGlobalStylesVariation,
+	premiumBadge,
+	showOnlyHoverViewDefaultVariation,
+	onSelect,
+}: GlobalStylesVariationsProps ) => {
+	const baseGlobalStyles = useMemo(
+		() =>
+			globalStylesVariations.find( ( globalStylesVariation ) =>
+				isDefaultGlobalStyleVariationSlug( globalStylesVariation )
+			) ?? ( {} as GlobalStylesObject ),
+		[ globalStylesVariations ]
+	);
+
+	const globalStylesVariationsWithoutDefault = useMemo(
+		() =>
+			globalStylesVariations.filter(
+				( globalStylesVariation ) => ! isDefaultGlobalStyleVariationSlug( globalStylesVariation )
+			),
+		[ globalStylesVariations ]
+	);
+
+	return (
+		<GlobalStylesContext.Provider value={ { base: baseGlobalStyles } }>
+			<div className="global-styles-variations">
+				<GlobalStylesVariation
+					key="base"
+					globalStylesVariation={ baseGlobalStyles }
+					isActive={ ! selectedGlobalStylesVariation }
```

Comment:
I found the default variation cannot be selected. Maybe we have to use `isDefaultGlobalStyleVariationSlug` here.

https://user-images.githubusercontent.com/13596067/228115420-56c1b37c-efcd-4da1-8095-9fe4b7ad963f.mov

Reply:
Good catch! For now, I'm supporting both scenarios: (1) when `selectedGlobalStyleVariation` is null or (2) when `selectedGlobalStylesVariation.slug` is default. We can make the behavior more consistent later 🙂